proxy/src/main/java/pt/webdetails/cda/CdaUtils.java

```diff
@@ -0,0 +1,611 @@
+/*!
+ * Copyright 2018 Webdetails, a Hitachi Vantara company. All rights reserved.
+ *
+ * This software was developed by Webdetails and is provided under the terms
+ * of the Mozilla Public License, Version 2.0, or any later version. You may not use
+ * this file except in compliance with the license. If you need a copy of the license,
+ * please go to  http://mozilla.org/MPL/2.0/. The Initial Developer is Webdetails.
+ *
+ * Software distributed under the Mozilla Public License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. Please refer to
+ * the license for the specific language governing your rights and limitations.
+ */
+
+package pt.webdetails.cda;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.StreamingOutput;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_FORM_URLENCODED;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.APPLICATION_XML;
+
+import pt.webdetails.cpf.utils.MimeTypes;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Iterator;
+
+import com.sun.jersey.api.client.Client;
+import com.sun.jersey.api.client.ClientResponse;
+import com.sun.jersey.api.client.WebResource;
+import com.sun.jersey.api.client.config.ClientConfig;
+import com.sun.jersey.api.client.config.DefaultClientConfig;
+import com.sun.jersey.api.client.filter.HTTPBasicAuthFilter;
+import com.sun.jersey.api.json.JSONConfiguration;
+
+
+@Path( "/cda/api" )
+public class CdaUtils {
+  private static final Log logger = LogFactory.getLog( CdaUtils.class );
+
+  private static String CDA_HOST = System.getProperty( "cda.hostname" );
+  private static String CDA_PORT = System.getProperty( "cda.port" );
+  private static String USER = System.getProperty( "repos.user" );
+  private static String PASS = System.getProperty( "repos.password" );
+  private String url_do_query = "http://" + CDA_HOST + ":"
+      + CDA_PORT + "/cxf/cda/cda/api/utils/doQuery";
+
+  public CdaUtils() {
+    String serverFlag = System.getProperty( "cda.to_server" );
+    if ( serverFlag != null && serverFlag.equals( "1" ) ) {
+      url_do_query = "http://" + CDA_HOST + ":" + CDA_PORT + "/pentaho/plugin/cda/api/doQuery";
+    }
+  }
+
+  @POST
+  @Path( "/doQuery" )
```

Comment:
I'm concerned about the scalability of this http proxying when we introduce the real time capabilities... We'll need to tunnel web socket connections, keeping them opened to both sides.

Would it be enough to override the `getPluginBase` (there is one in `Dashboard.ext.js`) and let the browser access  the CDA directly?